redfish-core/include/event_service_manager.hpp

```diff
@@ -1071,6 +1071,9 @@ class EventServiceManager
                     2, ' ', true, nlohmann::json::error_handler_t::replace);
                 entry->sendEvent(strMsg);
                 eventId++; // increament the eventId
+#ifdef BMCWEB_ENABLE_IBM_MANAGEMENT_CONSOLE
+                bmcweb::flightrecorder::FlightRecorder::GetInstance().saveRecord(msgJson);
```

Comment:
this should be moved to a handler that is called after a successful sendEvent and a callback that is called after unsuccessful retries finish.